android/app/src/main/java/org/smartregister/fhircore/activity/CqlActivity.kt

```diff
@@ -0,0 +1,187 @@
+import org.cqframework.cql.elm.execution.Library
+import org.cqframework.cql.elm.execution.VersionedIdentifier
+import org.hl7.fhir.instance.model.api.IBaseBundle
+import org.hl7.fhir.instance.model.api.IBaseResource
+import org.opencds.cqf.cql.engine.data.CompositeDataProvider
+import org.opencds.cqf.cql.engine.data.DataProvider
+import org.opencds.cqf.cql.engine.fhir.converter.FhirTypeConverter
+import org.opencds.cqf.cql.engine.fhir.converter.FhirTypeConverterFactory
+import org.opencds.cqf.cql.engine.fhir.model.R4FhirModelResolver
+import org.opencds.cqf.cql.evaluator.CqlEvaluator
+import org.opencds.cqf.cql.evaluator.cql2elm.content.LibraryContentProvider
+import org.opencds.cqf.cql.evaluator.cql2elm.content.fhir.BundleFhirLibraryContentProvider
+import org.opencds.cqf.cql.evaluator.cql2elm.util.LibraryVersionSelector
+import org.opencds.cqf.cql.evaluator.engine.execution.TranslatingLibraryLoader
+import org.opencds.cqf.cql.evaluator.engine.retrieve.BundleRetrieveProvider
+import org.opencds.cqf.cql.evaluator.engine.terminology.BundleTerminologyProvider
+import org.opencds.cqf.cql.evaluator.fhir.adapter.r4.AdapterFactory
+import org.opencds.cqf.cql.evaluator.library.CqlFhirParametersConverter
+import org.opencds.cqf.cql.evaluator.library.LibraryEvaluator
+import org.smartregister.fhircore.FhirApplication
+import org.smartregister.fhircore.R
+
+class CqlActivity : AppCompatActivity() {
+  private lateinit var fhirEngine: FhirEngine
+  private var fhirContext = FhirContext.forCached(FhirVersionEnum.R4)
+  private var adapterFactory = AdapterFactory()
+  private var fhirTypeConverter: FhirTypeConverter? = null
+  private var cqlFhirParametersConverter: CqlFhirParametersConverter? = null
+  private var libraryVersionSelector: LibraryVersionSelector? = null
+  private var contentProvider: LibraryContentProvider? = null
+  private var terminologyProvider: BundleTerminologyProvider? = null
+  private var bundleRetrieveProvider: BundleRetrieveProvider? = null
+  private lateinit var parser: IParser
+  private var cqlEvaluator: CqlEvaluator? = null
+  private var libraryEvaluator: LibraryEvaluator? = null
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    fhirEngine = FhirApplication.fhirEngine(this)
+
+    parser = fhirContext.newJsonParser()
+    parser.setPrettyPrint(true)
+    libraryVersionSelector = LibraryVersionSelector(adapterFactory)
+    fhirTypeConverter = FhirTypeConverterFactory().create(fhirContext.version.version)
+    cqlFhirParametersConverter =
+      CqlFhirParametersConverter(fhirContext, adapterFactory, fhirTypeConverter)
+    try {
+
+      val (ancLibrary, fhirHelperLibrary) =
+        runBlocking {
+          // Switch to background thread and continue blocking UI thread.
+          withContext(Dispatchers.IO) {
+            val ancLibrary = fhirEngine.load(org.hl7.fhir.r4.model.Library::class.java, "758")
```

Comment:
cool, to keep track of these for later replacement, how about we add a todo

```suggestion
            // todo: load ID from configuration file or elsewhere
            val ancLibrary = fhirEngine.load(org.hl7.fhir.r4.model.Library::class.java, "758")
```

Reply:
sure